examples/shader.rs

```diff
@@ -0,0 +1,194 @@
+use femtovg::renderer::OpenGl;
+use femtovg::{Canvas, Color, Paint, Path};
+use glow::{Context, HasContext, NativeFramebuffer, NativeProgram, NativeTexture, Program};
+use glutin::event::WindowEvent;
+use glutin::ContextBuilder;
+use glutin::{Api, GlRequest};
+use std::fs;
+use winit::event::Event;
+use winit::event_loop::{ControlFlow, EventLoop};
+use winit::window::WindowBuilder;
+
+const WINDOW_WIDTH: f32 = 640.0;
+const WINDOW_HEIGHT: f32 = 480.0;
+
+fn main() {
+    let el = EventLoop::new();
+
+    let windowed_context = {
+        let wb = WindowBuilder::new()
+            .with_inner_size(winit::dpi::PhysicalSize::<f32>::new(WINDOW_WIDTH, WINDOW_HEIGHT))
+            .with_resizable(false)
+            .with_decorations(true)
+            .with_title("SHADER");
+
+        let windowed_context = ContextBuilder::new()
+            .with_gl(GlRequest::Specific(Api::OpenGlEs, (3, 0)))
+            .with_vsync(false)
+            .build_windowed(wb, &el)
+            .unwrap();
+        let windowed_context = unsafe { windowed_context.make_current().unwrap() };
+        windowed_context
+    };
+
+    let context: Context;
+    let framebuffer: NativeFramebuffer;
+    let texture_colorbuffer: NativeTexture;
+    let shader_program: Program;
+    let mut renderer: OpenGl;
+
+    unsafe {
+        context = glow::Context::from_loader_function(|symbol| windowed_context.get_proc_address(symbol) as *const _);
+        renderer = OpenGl::new_from_function(|symbol| windowed_context.get_proc_address(symbol) as *const _).unwrap();
+
+        shader_program = create_shader_program(&context);
+        (framebuffer, texture_colorbuffer) = create_framebuffer_colorbuffer(&context);
+
+        renderer.set_screen_target(Some(framebuffer));
+    }
+
+    let mut canvas = Canvas::new(renderer).expect("Cannot create canvas");
+    canvas.set_size(WINDOW_WIDTH as u32, WINDOW_HEIGHT as u32, 1.0);
+
+    el.run(move |event, _, control_flow| {
+        *control_flow = ControlFlow::Poll;
+
+        match event {
+            Event::LoopDestroyed => return,
+            Event::WindowEvent { ref event, .. } => match event {
+                WindowEvent::CloseRequested => *control_flow = ControlFlow::Exit,
+                _ => {}
+            },
+            Event::RedrawRequested(_) => {
+                // draw red rectangle on white background
+
+                let dpi_factor = windowed_context.window().scale_factor();
+                let size = windowed_context.window().inner_size();
+                canvas.set_size(size.width as u32, size.height as u32, dpi_factor as f32);
+                canvas.clear_rect(0, 0, size.width as u32, size.height as u32, Color::rgbf(1., 1., 1.));
+
+                prepare_framebuffer_for_render(&context, framebuffer);
```

Comment:
Suggestion to swap things around, so that prepare_framebuffer_for_render() prepares the scene for use with the femtovg canvas, which then does its thing, ends with `flush()` and then `render_framebuffer_to_screen` takes over.

```suggestion
                prepare_framebuffer_for_render(&context, framebuffer);

                // draw red rectangle on white background

                let dpi_factor = windowed_context.window().scale_factor();
                let size = windowed_context.window().inner_size();
                canvas.set_size(size.width as u32, size.height as u32, dpi_factor as f32);
                canvas.clear_rect(0, 0, size.width as u32, size.height as u32, Color::rgbf(1., 1., 1.));
```